New util: xray near energy

vidartf opened the pull request:
Added a utility function to lookup x-ray lines near a given energy. Useful for matching unknown EDS peaks.

### TODO
- [x] Tests
- [x] Documentation

pburdet commented on hyperspy/misc/eds/utils.py:
```diff
@@ -13,11 +13,44 @@ def _get_element_and_line(Xray_line):
+        try:
+            lines = el_props['Atomic_properties']['Xray_lines']
+        except KeyError:
+            continue
```
A question of style, something like that works as well

```python
    
if 'Xray_lines' in el_props['Atomic_properties']:
    lines = el_props['Atomic_properties']['Xray_lines']
```

vidartf replied:
Yes, as long as they all have `Atomic_properties`. If they do (and always will), I'll make the change.

pburdet replied:
The database has been so far build for EELS and EDS purpose. Elements always have atomic properties. Anyway, minor issue. Your version is more stable

pburdet commented on hyperspy/misc/eds/utils.py:
```diff
@@ -13,11 +13,44 @@ def _get_element_and_line(Xray_line):
+    element, line = _get_element_and_line(xray_line)
+    return elements_db[element]['Atomic_properties']['Xray_lines'][
         line]['energy (keV)']
 
```
Thanks for the correction

pburdet commented:
Tests and documentation are needed

vidartf commented:
I'm struggling a bit to figure out where is the best spot to put the docs: `utils.rst` or `eds.rst`? Also, I'm unsure if the function should be added to `utils/eds.py` or not.

vidartf commented:
Moved `_signals.eds._parse_only_lines()` to `misc.eds.utils`, as it was independent of any instance, and it was also useful elsewhere. As it is an internal function that is quite new, the move should have no other repercussions at the moment. Also added relevant tests.

pburdet commented:
I would put the documentation in `eds.rst`. Ithink it's more logic.

I'm all for adding it in `utils.eds`, this is where all utility functions for eds should be.

vidartf commented:
Docs added. Looks OK?

francisco-dlp commented on hyperspy/misc/eds/utils.py:
```diff
@@ -13,11 +13,62 @@ def _get_element_and_line(Xray_line):
+    return only_lines
+
+
+def get_xray_lines_near_energy(energy, spread=0.1, only_lines=None):
```
Wouldn't an 'radius' be more explicit and adequate than `spread`?

`width`, `size` or `length` could also be good alternatives, but would require

    E_min, E_max = energy - spread/2., energy + spread/2.

vidartf replied:
Makes sense. I struggled to come up with a good argument name, so good input. I think `width` makes the most sense to me, or maybe even better: `window_width`?

francisco-dlp replied:
`width`, `window_width` or `interval_width` all make sense to me. `width` has the advantage of being shorter.

vidartf replied:
Ended up using `width`, while referring to it as a window width in the docstrings, that was also updated.

vidartf commented:
PS: I don't know how 996c98d is marked as passing the test before my test fix. If you click on the log, it actually says it has one failure...
